flows/vision-zero/cris_import/cris_import.py

```diff
@@ -0,0 +1,378 @@
+        logger.info(command)
+        cmd = command.split()
+        rm_result = Popen(cmd, stdout=PIPE, stderr=PIPE, stdin=PIPE).stdout.read()
+        logger.info(rm_result)
+
+    return None
+
+
+with Flow(
+    "CRIS Crash Import",
+    # schedule=Schedule(clocks=[CronClock("* * * * *")]),
+    run_config=UniversalRun(labels=["vision-zero", "atd-data03"]),
+    state_handlers=[skip_if_running_handler],
+) as flow:
+    # make sure we have the docker image we want to use to process these built.
+    image = build_docker_image()
+
+    # get a location on disk which contains the zips from the sftp endpoint
+    zip_location = download_extract_archives()
+
+    # iterate over the zips in that location and unarchive them into
+    # a list of temporary directories containtain the files of each
+    extracted_archives = unzip_archives(zip_location)
+
+    # push up the archives to s3 for archival
+    uploaded_archives_csvs = upload_csv_files_to_s3.map(extracted_archives)
+
+    # def run_docker_image(extracted_data, vz_etl_image, command):
+    crash_import_tmpdirs = run_docker_image.map(
+        extracted_data=uploaded_archives_csvs,
+        vz_etl_image=unmapped(image),
+        command=unmapped(["/app/process_hasura_import.py", "crash"]),
+    )
+
+    # the next four set of task calls; i want them to be in parallel, but this won't work without a dask setup, i think
+    unit_import_tmpdirs = run_docker_image.map(
+        extracted_data=uploaded_archives_csvs,
+        vz_etl_image=unmapped(image),
+        command=unmapped(["/app/process_hasura_import.py", "unit"]),
+    )
+    unit_import_tmpdirs.set_upstream(crash_import_tmpdirs)
+
+    person_import_tmpdirs = run_docker_image.map(
+        extracted_data=uploaded_archives_csvs,
+        vz_etl_image=unmapped(image),
+        command=unmapped(["/app/process_hasura_import.py", "person"]),
+    )
+    person_import_tmpdirs.set_upstream(crash_import_tmpdirs)
+
+    primaryperson_import_tmpdirs = run_docker_image.map(
+        extracted_data=uploaded_archives_csvs,
+        vz_etl_image=unmapped(image),
+        command=unmapped(["/app/process_hasura_import.py", "primaryperson"]),
+    )
+    primaryperson_import_tmpdirs.set_upstream(crash_import_tmpdirs)
+
+    charges_import_tmpdirs = run_docker_image.map(
+        extracted_data=uploaded_archives_csvs,
+        vz_etl_image=unmapped(image),
+        command=unmapped(["/app/process_hasura_import.py", "charges"]),
+    )
+    charges_import_tmpdirs.set_upstream(crash_import_tmpdirs)
+
+    # remove archives from SFTP endpoint
+    removal_token = remove_archives_from_sftp_endpoint(zip_location)
+    removal_token.set_upstream(crash_import_tmpdirs)
+    removal_token.set_upstream(unit_import_tmpdirs)
+    removal_token.set_upstream(person_import_tmpdirs)
+    removal_token.set_upstream(primaryperson_import_tmpdirs)
+    removal_token.set_upstream(charges_import_tmpdirs)
+
+    cleanup = cleanup_temporary_directories(
```

Comment:
Also wanted to note that I was happy to hear that some the processes that would be hard to know were even running were pulled into this flow. Very cool to make them more visible as part of this port.

Reply:
@mddilley wrote: 
> Was this the part that you were referring that was running elsewhere? @frankhereford

Yep, you are correct. I'm referring to the cronjob that runs on the SFTP endpoint that daily unzips any zip files it's received and uploads the resulting CSV files to airflow. This was not part of the airflow ETL, but was a behavior that the airflow ETL depended on. 

It's because I short-circuit the uploading of CSV files by the SFTP endpoint, that this flow takes on the responsibility of uploading the CSV files to S3 for archival -- so we do get that permanent copy which we are used to.